### DISCUSSION TITLE
What's the "right" way to disable completion until I press ctrl-space?

### DISCUSSION THREAD

**Discussion by agriffis** [saghen/blink.cmp#563] [Q&A] [answered]
I'd like to disable autocompletion until I press a key. I don't mind if it's looking things up in the background, but I don't want stuff happening on the screen until I ask for it.

I'm trying this, but I'd love some confirmation that I'm on the right track:

```
      -- Don't show completion menu until I press ctrl-space.
      completion = {
        trigger = {
          show_on_keyword = false,
          show_on_trigger_character = false,
        }
      },
```

**Comment by saghen** [comment, marked as answer]
`completion.menu.auto_show = false` would still request the completions in the background but won't show the menu until you open it explicitly. You might want to try `ghost_text` with this configuration fyi

**Reply by agriffis** [reply]
Oh, cool! If I want a quick way to accept the ghost text without opening the menu, is that possible? It looks like `select_and_accept` checks if the menu is open: https://github.com/Saghen/blink.cmp/blob/c8e86a3ed1eff07e2c1108779a720d3b4c6b86a7/lua/blink/cmp/init.lua#L67-L80

**Reply by agriffis** [reply]
I guess this works:

```
      keymap = {
        ['<Tab>'] = {
          function(cmp)
            local completion_list = require('blink.cmp.completion.list')
            vim.schedule(function()
              completion_list.accept {
                index = completion_list.selected_item_idx or 1,
              }
            end)
            return true
          end,
        },
      },
```

**Reply by saghen** [reply]
Ah yep, I have a todo to check if ghost text is active in `cmp.is_visible()`. I'll try to remember to ping here when that's added :)

**Reply by saghen** [reply]
Added a check for ghost text in https://github.com/Saghen/blink.cmp/commit/1006662ad53c92adf9ae6f2d05cee38f613d08ff
